### ISSUE TITLE
as.IDate error that can be fixed

### ISSUE THREAD

**Issue by arunsrinivasan** [Rdatatable/data.table#4676]
```r
require(data.table) # 1.13.1 from today
x <- c("2020-01-01", "")
y <- c("", "2020-01-01")
as.IDate(x) # works fine
as.IDate(y)
# Error in charToDate(x) : 
#   character string is not in a standard unambiguous format
```

**Comment by MichaelChirico** [comment]
Thanks @arunsrinivasan , this is inherited from `Date`, so to get around it, we'd need to build our own parser basically:

```
require(data.table) # 1.13.1 from today
x <- c("2020-01-01", "")
y <- c("", "2020-01-01")
as.Date(x) # works fine
as.Date(y) # same issue
```

**Comment by arunsrinivasan** [comment]
I disagree. We should issue a fix while filing an issue there. Like I said, it takes time to get things into base R and people don't always use the most recent version of R, if you use in production. I'm all up for filing an issue there, but I'd like the issue taken care of in data.table nonetheless.

**Comment by MichaelChirico** [comment]
I agree with you about it being easier/more common to update `data.table` than R, and agree we should make a fix in `data.table` no matter what, I am just pushing back on doing so immediately.

What I'm trying to prevent is setting ourselves up for a breaking change in a later release:

T0: we submit an issue to Bugzilla & merge our own solution in `data.table`
T1: release this version of `data.table`
T2: R disagrees with our solution, but _does_ fix the issue in a different, incompatible way
T3: Because we inherit from/extend `Date`, we have to patch our original fix [breaking change]

So I think we should at least file the issue first & see if R core has any opinion. If not we just merge & move on.

**Comment by MichaelChirico** [comment]
I have submitted a bug to R:

https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17909

Note the simpler fix:

```
is.na(x) <- !nzchar(x)
```

**Comment by arunsrinivasan** [comment]
> T2: R disagrees with our solution, but does fix the issue in a different, incompatible way

Sorry but I'm having a hard time making sense of this. There are only two possible scenarios I can think of.

1. `as.Date(x)` is the same as `as.Date(x, format="%Y-%m-%d")`, in which case it doesn't matter how you get there, at least temporarily until base R accepts your PR or comes up with another.

2. They keep the *error*, in which case, we all agree `data.table` could and should handle this.

It has to be equal to the output of `as.Date(x, format=)` in a non-error scenario. What other *incompatible* ways are there?

**Comment by mmaechler** [comment]
> I have submitted a bug to R:
> 
> https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17909

I've fixed the bug in  R  now  (svn rev 79019 for 'R-devel'  -- to be ported to  "R 4.0.2 patched" in a few days ("almost surely").

**Comment by ColeMiller1** [comment]
R fixed this in 4.0.3 released on October 10. Should this be closed as is or should Michael's PR be merged first?

**Comment by MichaelChirico** [comment]
would be nice to have something that works on our dependency ( R >= 3.1.0 )
